CMP1903M Object Oriented Programming A01 2223/CMP1903M A01 2223/CMP1903M A01 2223/Pack.cs

```diff
@@ -9,56 +9,66 @@ namespace CMP1903M_A01_2223
 {
     class Pack
     {
-        
-        //Get, set values
-        public int typeOfShuffle { get; set; }
-        public int amount { get; set; }
-
         //Initialise the card pack here
-        public static List<Card> _pack = new List<Card>();
-        public Card[] cardPack = _pack.ToArray();
-
+        private static Card[] _pack;
         public Pack()
         {
+            _pack = new Card[52];
+            CreatePack();
         }
-        public static bool shuffleCardPack(int typeOfShuffle, int[] cardPack)
+
+        private static void FisherShuffle(ref Card[] _pack)
+        {
+            Console.WriteLine("Fisher–Yates shuffle");
+            Random _random = new Random();
+            int n = _pack.Length;
+            Card[] tempArr = new Card[n];
+
+            //Will change all positions of cards with a random card
+            for (int i = 0; i < n; i++)
+            {
+                int r = _random.Next(n);
+                if (r != i) tempArr[i] = tempArr[r];
+                tempArr[r] = _pack[i];
+            }
+            _pack = tempArr;
+            
+        }
+
+        private static void RiffleShuffle(ref Card[] _pack)
+        {
+            Console.WriteLine("Tabled Card Riffle Shuffle");
+            int n = _pack.Length;
+
+            //Creates two halves of the deck 
+            Card[] left = new Card[26];
+            Card[] right = new Card[26];
+            Card[] shuffledCards = new Card[52];
+
+            //Shuffles with both arrays from both ends
+            for (int i = 0; i < (n / 2); i++)
+            {
+                _pack[i] = left[i];
+                _pack[26 + i] = right[i];
+            }
+            for (int i = 0; i < 26; i++)
+            {
+                shuffledCards[i] = left[i];
+                shuffledCards[i + 1] = right[i];
+                i++;

```

Comment:
Not a big issue but for code clarity sake you could replace the i++ here and put in the for loop statement itself by doing something like for(int i = 0; i < 26; i = i + 2)
As you are essentially just looping through the for loop and every time adding 2 instead of 1 to your iterator